driver-core/src/main/com/mongodb/client/model/changestream/TruncatedArray.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.changestream;
+
+import org.bson.codecs.pojo.annotations.BsonCreator;
+import org.bson.codecs.pojo.annotations.BsonProperty;
+
+import java.util.Objects;
+
+import static com.mongodb.assertions.Assertions.isTrueArgument;
+import static com.mongodb.assertions.Assertions.notNull;
+
+/**
+ * A part of an {@link UpdateDescription} object specifying a change to a field of the {@linkplain org.bson.BsonType#ARRAY array} type
+ * when the change is reported as truncation.
+ *
+ * @since 4.3
+ */
+public final class TruncatedArray {
+    private final String field;
+    private final int newSize;
+
+    /**
+     * Not part of the public API.
```

Comment:
I'm not sure this is the case.

Reply:
I think this would be the first time we've tried to make a single method of an otherwise public class not part of the public API.  I'm not sure I like the precedent.

Reply:
This is why I asked if users are supposed to create instances of `UpdateDescription` (a part of `ChangeStreamDocument`) on their own. Since the answer was "no", I concluded that constructors of these classes are not part of the public API. If that is not the case, then then the answer to the mentioned question is "yes", i.e., users are supposed to create instances of these classes, including the new `TruncatedArray` class (@rozza pointed out a [good example](https://github.com/mongodb/mongo-java-driver/pull/667#discussion_r576681615) when this may be needed).

Thank you, guys, I will make the changes necessary to express that the constructors are a part of public API.

Reply:
Done.